backend/strapi-sync/sync.js

```diff
@@ -0,0 +1,404 @@
+      page += 1;
+
+      let paginatedUrl = currentUrl;
+
+      if (queryParams) {
+        paginatedUrl = `${paginatedUrl}&pagination[page]=${page}`;
+      } else {
+        paginatedUrl = `${paginatedUrl}?pagination[page]=${page}`;
+      }
+
+      const pageData = await getPageData(paginatedUrl);
+
+      items.push(...pageData);
+    }
+
+    return items;
+  } catch (error) {
+    console.error(error);
+    return [];
+  }
+}
+
+export async function fetchAllModels() {
+  const url = "https://cms.bcparks.ca/api";
+
+  const strapiData = [
+    {
+      endpoint: "/park-operations",
+      model: "park-operation",
+      fields: null,
+      items: [],
+    },
+    {
+      endpoint: "/park-operation-sub-areas",
+      model: "park-operation-sub-area",
+      fields: ["protectedArea", "parkSubAreaType"],
+      items: [],
+    },
+    {
+      endpoint: "/protected-areas",
+      model: "protected-area",
+      fields: ["parkFacilities", "parkOperations"],
+      items: [],
+    },
+    {
+      endpoint: "/camping-types",
+      model: "camping-type",
+      fields: null,
+      items: [],
+    },
+    {
+      endpoint: "/facility-types",
+      model: "facility-type",
+      fields: null,
+      items: [],
+    },
+    {
+      endpoint: "/park-operation-sub-area-types",
+      model: "park-operation-sub-area-type",
+      fields: ["facilityType", "campingType"],
+      items: [],
+    },
+    {
+      endpoint: "/park-operation-dates",
+      model: "park-operation-date",
+      fields: ["protectedArea"],
+      items: [],
+    },
+    {
+      endpoint: "/park-operation-sub-area-dates",
+      model: "park-operation-sub-area-date",
+      fields: ["parkOperationSubArea"],
+      items: [],
+    },
+  ];
+
+  await Promise.all(
+    strapiData.map(async (item) => {
+      const currentUrl = url + item.endpoint;
+      let qpString = null;
+
+      if (item.fields) {
+        const queryParams = [];
+
+        for (const field of item.fields) {
+          const param = `populate[${field}][fields]=id`;
+
+          queryParams.push(param);
+        }
+
+        qpString = queryParams.join("&");
+      }
+
+      const temp = getData(currentUrl, qpString);
+
+      item.items = temp;
+    }),
+  );
+
+  return strapiData;
+}
+
+export async function createOrUpdatePark(item) {
+  // get park on our DB by strapi ID
+  let dbItem = await getItemByAttributes(Park, { strapiId: item.id });
+
+  if (dbItem) {
+    dbItem.name = item.attributes.protectedAreaName;
+  } else {
+    const dateable = await createModel(Dateable);
+    const data = {
+      name: item.attributes.protectedAreaName,
+      orcs: item.attributes.orcs,
+      dateableId: dateable.id,
+      strapiId: item.id,
+    };
+
+    dbItem = await createModel(Park, data);
+  }
+
+  return dbItem;
+}
+
+export async function syncParks(parkData) {
+  const items = parkData.items;
+
+  items.forEach(async (item) => {
+    await createOrUpdatePark(item);
+  });
```

Comment:
An async function implicitly returns a Promise, so `forEach` won't actually wait for the calls to `createOrUpdatePark` to finish here and it will run them all at the same time.  If we really want to `await` and run each call one-by-one, we can change this to a `for..of` loop. If we _do_ want to run them all in parallel, we could take out the `async`/`await`, but in that case we should use `Promise.all` so we can catch exceptions.

Looks like this has a decent explanation if you're curious: https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971